sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +764,83 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Initiates a search and returns a {@link PhoneNumberSearch} usable by other functions
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param options A {@link CreateSearchOptions} with the search options
+     * @param pollInterval The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the search result
+     */
+    public PollerFlux<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
+        CreateSearchOptions options, Duration pollInterval) {
+        Objects.requireNonNull(options, "'options' cannot be null.");
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(pollInterval,
+            createSearchActivationOperation(options),
+            createSearchPollOperation(),
+            cancelSearchOperation(),
+            createSearchFetchResultOperation());
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PhoneNumberSearch>> createSearchActivationOperation(CreateSearchOptions options) {
+        return (pollingContext) -> {
+            Mono<PhoneNumberSearch> response = createSearch(options).flatMap(createSearchResponse -> {
+                String searchId = createSearchResponse.getSearchId();
+                Mono<PhoneNumberSearch> phoneNumberSearch = getSearchById(searchId);
+                return phoneNumberSearch;
+            });
+            return response;
+        };
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PollResponse<PhoneNumberSearch>>> createSearchPollOperation() {
+        return pollingContext ->
+            getSearchById(pollingContext.getActivationResponse().getValue().getSearchId())
```

Comment:
Can't we do something like pollingContext.getLatestResponse().. this is pretty much getting the searchId and querying the sync status api

Reply:
The activation response has the search id I'm trying to create, so to poll on the status on that number, I need to get that searchId from the activation operation. The getLatestResponse() function just returns the latest from the polling operation, so at that moment, getLatestResponse() doesn't have anything